core-bundle/src/DependencyInjection/Compiler/AddCronJobsPass.php

```diff
@@ -53,14 +59,28 @@ public function process(ContainerBuilder $container): void
                     throw new InvalidDefinitionException(sprintf('The contao.cronjob definition for service "%s" has an invalid interval expression "%s"', $serviceId, $interval));
                 }
 
-                $definition->addMethodCall(
-                    'addCronJob',
-                    [
-                        new Definition(CronJob::class, [new Reference($serviceId), $interval, $method]),
-                    ]
-                );
+                $newDefinition = new Definition(CronJob::class, [new Reference($serviceId), $interval, $method]);
+
+                $reflector = new \ReflectionMethod($jobDefinition->getClass(), $method ?? '__invoke');
+                $returnType = $reflector->getReturnType();
+                $returnsPromise = $returnType instanceof \ReflectionNamedType && PromiseInterface::class === $returnType->getName();
```

Comment:
```suggestion
                $returnsPromise = $returnType instanceof \ReflectionNamedType && is_a($returnType->getName(), PromiseInterface::class, true);
```
This would allow for covariance i.e. a more specific return type.